core/src/main/scala/kafka/coordinator/transaction/TransactionMetadata.scala

```diff
@@ -490,9 +494,10 @@ private[transaction] class TransactionMetadata(val transactionalId: String,
           }
 
         case CompleteAbort | CompleteCommit => // from write markers
+          // With transaction V2, we allow Empty transaction to be aborted, so the txnStartTimestamp can be -1.
```

Comment:
I wonder if we can have a way to check/only allow -1 timestamp if the previous state was empty 🤔

Reply:
I just mean if there was a way to enforce that only that case is -1 here in the code.

Reply:
Added the check for the current state.

Reply:
The txnStartTimestamp is updated at the beginning of the transaction, not the end of the transaction.

Reply:
I wonder what the implication is when we start the transaction with an abort 😅

Reply:
I can use the update time to be the txn start time. I guess it is better than make more exceptions in the state transition which confuses the state machine.